src/modules/tools/zprobe/ZGridStrategy.cpp

```diff
@@ -534,8 +543,22 @@ void ZGridStrategy::normalize_grid()
 
 void ZGridStrategy::homexyz()
 {
-    Gcode gc("G28", &(StreamOutput::NullStream));
-    THEKERNEL->call_event(ON_GCODE_RECEIVED, &gc);
+
+  switch(this->home_before_probe) {
+    case NOHOME : return;
+
+    case HOMEXY : {
+        Gcode gc("G28 X Y", &(StreamOutput::NullStream));
```

Comment:
This was correct before it must be G28 X0 Y0. X Y is illegal gcode and we are trying to keep out GCode to spec. This is not marlin!